packages/core/src/components/tag/tag.tsx

```diff
@@ -135,7 +135,12 @@ export const Tag: React.FC<TagProps> = React.forwardRef((props, ref) => {
     );
 
     return (
-        <span {...removeNonHTMLProps(htmlProps)} {...interactiveProps} className={tagClasses}>
+        <span
+            {...removeNonHTMLProps(htmlProps)}
+            {...interactiveProps}
+            className={tagClasses}
+            role={interactiveProps.onClick !== undefined ? "button" : undefined}
```

Comment:
Can we reuse the `isInteractive` variable from above? Seems like it would read a bit more directly here:
```suggestion
            role={isInteractive ? "button" : undefined}
```